.github/workflows/ci.yaml

```diff
@@ -0,0 +1,66 @@
+name: CI
+run-name: ${{ github.actor }} is running CI
+
+# workflow_dispatch allows you to run this workflow manually from the Actions tab
+on: [ pull_request, workflow_dispatch ]
+
+# This allows a subsequently queued workflow run to interrupt previous runs
+concurrency:
+  group: ${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+jobs:
+  tests:
+    name: Tests
+    runs-on: 'ubuntu-20.04'
+    strategy:
+      matrix:
+        include:
+          - php-versions: '8.2'
+            phpunit-versions: '9.6'
+          - php-versions: '8.2'
+            phpunit-versions: '10.0'
+          - php-versions: '8.2'
+            phpunit-versions: 'latest'
+          - php-versions: '8.1'
+            phpunit-versions: '9.6'
+          - php-versions: '8.1'
+            phpunit-versions: '10.0'
+          - php-versions: '8.1'
+            phpunit-versions: 'latest'
+          - php-versions: '8.0'
+            phpunit-versions: '9.6'
+          - php-versions: '7.4'
+            phpunit-versions: '9.6'
+          - php-versions: '7.3'
+            phpunit-versions: '9.6'
```

Comment:
I think there is no point to support EOL versions of PHP also I would not recommend to support multiple versions of PHPUnit - it's dev dependency so I would just use the latest one, and with that your test matrix will be just half.

I would even drop support for PHP 8.0 because it's EOL later this year + then you only need that one version of PHPUnit

Reply:
Very true. Bumped the minimum versions to php 8.1 and phpunit 10.0.